server/src/main/java/io/spine/server/model/HandlerMethodFailedException.java

```diff
@@ -59,6 +69,40 @@ public HandlerMethodFailedException(Object  target,
          */
         this.dispatchedMessage = (GeneratedMessageV3) checkNotNull(dispatchedMessage);
         this.messageContext = (GeneratedMessageV3) checkNotNull(messageContext);
+        setProducer(cause, target);
+    }
+
+    /**
+     * If the root cause of the exception was thrown by a handler method,
+     * {@linkplain ThrowableMessage#initProducer(Any) sets the identity} of the object which thrown.
+     */
+    private static void setProducer(Exception exception, Object target) {
+        Throwable rootCause = getRootCause(exception);
+        if (rootCause instanceof ThrowableMessage) {
+            ThrowableMessage thrownMessage = (ThrowableMessage) rootCause;
+            Any producerId = idOf(target);
+            thrownMessage.initProducer(producerId);
+        }
+    }
+
+    /**
+     * Obtains an identity of an object which thrown {@link ThrowableMessage}.
+     *
+     * @implNote Attempts to cast the passed object to {@link Entity} or {@link EventProducer}.
+     * If none of this works, returns the result of {@link Object#toString()}.
+     */
+    @SuppressWarnings("ChainOfInstanceofChecks")
+    public static Any idOf(Object target) {
+        if (target instanceof Entity) {
+            Object entityId = ((Entity) target).getId();
+            return Identifier.pack(entityId);
+        }
+
+        if (target instanceof EventProducer) {
+            return ((EventProducer) target).getProducerId();
+        }
+
+        return TypeConverter.toAny(target.toString());
```

Comment:
Let's use `Identifiers.pack(...)`. There are no differences between the results of those two, but it feels better to use `Identifiers` for IDs.